### PR TITLE
Update nrfxlib to be compatible with jan.5 Zephyr

### PR DESCRIPTION
**Pull request by SebastianBoe**
Changes required for compatibility with new Zephyr revision.

See commit messages for details

### CODE REVIEW

**Review by frkv — APPROVED** [review]
LGTM

File: zboss/src/CMakeLists.txt [hunk 1]
```diff
@@ -13,6 +13,13 @@ zephyr_library_include_directories(
     include
   )
 
+# Prevent the -Waddress-of-packed-member warning. We appear to be
+# taking addresses of struct fields in a manner that is not
+# safe. Until this is resolved upstream, we temporarily remove this
+# warning.
+zephyr_library_compile_options(-Wno-address-of-packed-member)
+
+
```
Comment (tejlmand) [inline zboss/src/CMakeLists.txt:22]: minor nit: why the extra empty line ?
```suggestion
```

**Review by tejlmand — APPROVED** [review]
lgtm.

minor nit noticed.

File: zboss/src/CMakeLists.txt (hunk 1 above)
Comment (SebastianBoe) [reply zboss/src/CMakeLists.txt:22]: Accidental.

Would prefer to not do any git paperwork for this, but will fix if I have to touch the code any way.

File: nrf_security/configs/legacy_crypto_config.h.template [hunk 2]
```diff
@@ -1906,16 +1630,6 @@
...
-/**
- * \def MBEDTLS_SSL_EXPORT_KEYS
- *
- * Enable support for exporting key block and master secret.
- * This is required for certain users of TLS, e.g. EAP-TLS.
- *
- * Comment this macro to disable support for key export
- */
-#define MBEDTLS_SSL_EXPORT_KEYS
-
```
Comment (lmaciejonczyk) [inline nrf_security/configs/legacy_crypto_config.h.template:1918]: The feature is needed for OpenThread. The OpenThread based samples don't compile without it.

**Review by lmaciejonczyk — DISMISSED** [review]

File: nrf_security/configs/legacy_crypto_config.h.template (hunk 2 above)
Comment (frkv) [reply nrf_security/configs/legacy_crypto_config.h.template:1918]: Will fix with proper configuration that is exposed

**Comment by lmaciejonczyk** [comment]
From OpenThread perspective, what have changed and makes the build failure is:

- missing MBEDTLS_SSL_EXPORT_KEYS in nrf-config.h
- now CONFIG_MBEDTLS_SSL_PROTO_DTLS=y needs to be explicitly enabled
- MBEDTLS_ENTROPY_C is disabled by default and cannot be y enabled because it's promptless option

**Comment by SebastianBoe** [comment]
@lmaciejonczyk : I believe MBEDTLS_SSL_EXPORT_KEYS has been fixed.

See https://github.com/nrfconnect/sdk-nrfxlib/pull/613/commits/7142b6ee1ca35e26541f935e5047100116d96d8b#diff-5381c07ad9e799f8bbf6967323abc7876f592fd24f9dab1f93822531f7b0136bR1618

**Comment by frkv** [comment]
> 

I would set the CONFIG_MBEDTLS_SSL_PROTO_DTLS either with a defconfig or by using the openthread security profile. Not everybody is using DTLS and it is uses more RAM and FLASH.

I don't know why you needed to set MBEDTLS_ENTROPY_C specifically. It is on a list of configurations that most likely will change due to PSA APIs becoming the standard one. If you are building with TF-M then the configuration is not applicable at all...

**Comment by lmaciejonczyk** [comment]
> @lmaciejonczyk : I believe MBEDTLS_SSL_EXPORT_KEYS has been fixed.
> 
> See [7142b6e#diff-5381c07ad9e799f8bbf6967323abc7876f592fd24f9dab1f93822531f7b0136bR1618](https://github.com/nrfconnect/sdk-nrfxlib/commit/7142b6ee1ca35e26541f935e5047100116d96d8b#diff-5381c07ad9e799f8bbf6967323abc7876f592fd24f9dab1f93822531f7b0136bR1618)

I believe we are still missing:
```kconfig_check_and_set_base(MBEDTLS_SSL_EXPORT_KEYS)``` in ```legacy_crypto_config.cmake```

**Comment by SebastianBoe** [comment]
@lmaciejonczyk :

> I believe we are still missing:
> kconfig_check_and_set_base(MBEDTLS_SSL_EXPORT_KEYS) in legacy_crypto_config.cmake
> 

I've now pushed this patch, good catch.

**Comment by lmaciejonczyk** [comment]
> I don't know why you needed to set MBEDTLS_ENTROPY_C specifically. It is on a list of configurations that most likely will change due to PSA APIs becoming the standard one. If you are building with TF-M then the configuration is not applicable at all...

OpenThread currently uses MBEDTLS_ENTROPY_C related function for initializing random number generator.

**Comment by frkv** [comment]
> > I don't know why you needed to set MBEDTLS_ENTROPY_C specifically. It is on a list of configurations that most likely will change due to PSA APIs becoming the standard one. If you are building with TF-M then the configuration is not applicable at all...
> 
> OpenThread currently uses MBEDTLS_ENTROPY_C related functions (mbedtls_entropy_init, mbedtls_entropy_func) for initializing random number generator.

This topic will be handled in the meeting at 14.00

**Review by tomchy — APPROVED** [review]

File: nrf_security/configs/legacy_crypto_config.h.template (hunk 2 above)
Comment (SebastianBoe) [reply nrf_security/configs/legacy_crypto_config.h.template]: fixed

File: crypto/nrf_oberon/include/ocrypto_curve_p224.h [hunk 3]
```diff
@@ -106,7 +106,7 @@ int ocrypto_curve_p224_scalarmult_base(ocrypto_cp_p224 *r, const ocrypto_sc_p224
  * @param       q       input point.
  *
  * @retval -1 if r = [0,0].
- * @retval 0 if successful.
+ * @etval 0 if successfull.
```
Comment (b-gent) [inline crypto/nrf_oberon/include/ocrypto_curve_p224.h:109]:
```suggestion
 * @retval 0 if successful.
```

File: crypto/nrf_oberon/include/ocrypto_ecdh_p256.h [hunk 4]
```diff
@@ -82,7 +90,7 @@ int ocrypto_ecdh_p256_common_secret(uint8_t r[32], const uint8_t s[32], const ui
  * @code
  *   ocrypto_ecdh_p256_common_secret_init(ctx, sKey, pKey);
  *   while (ocrypto_ecdh_p256_common_secret_iterate(ctx));
- *   res = ocrypto_ecdh_p256_common_secret_final(ctx, secret);
+ *   res = ocrypto_ecdh_p256_common_secret_final(ctx, secet);
```
Comment (b-gent) [inline crypto/nrf_oberon/include/ocrypto_ecdh_p256.h:93]:
```suggestion
 *   res = ocrypto_ecdh_p256_common_secret_final(ctx, secret);
```

File: crypto/nrf_oberon/include/ocrypto_rsa.h [hunk 5]
```diff
@@ -191,18 +192,18 @@ int ocrypto_rsa1024_oaep_sha256_crt_decrypt(
  *
  * @param[out] s       The generated 128-byte signature.
  * @param      m       The message to be signed.
- * @param      mlen    Length of @p m.
- * @param      k       A valid 1024-bit RSA secret key.
+ * @param      m_len   Length of @p m.
+ * @param      sk      A valid 1024-bit RSA secret key.
  *
- * @return 0
+ * @retval 0
```
Comment (b-gent) [inline crypto/nrf_oberon/include/ocrypto_rsa.h:198]:
```suggestion
 * @return 0 on success.
```
Using ``@retval`` only makes sense to present and explain several return values.
It's a nit and need not be changed, just saying.

**Review by b-gent — CHANGES_REQUESTED** [review]

**Comment by b-gent** [comment]
I had a brief look at doc changes, leaving a few comments.

**Comment by SebastianBoe** [comment]
@b-gent : Thank you, all feedback has been addressed

**Review by b-gent — APPROVED** [review]

**Comment by b-gent** [comment]
@SebastianBoe  thank you!

File: nrf_security/src/zephyr/CMakeLists.txt [hunk 6]
```diff
@@ -33,9 +35,12 @@ if(CONFIG_BUILD_WITH_TFM)
     )
   endif()
 
+  # For some reason $<TARGET_PROPERTY:tfm,TFM_BINARY_DIR> does not work here
```
Comment (SebastianBoe) [inline nrf_security/src/zephyr/CMakeLists.txt:38]: Probably due to this problem:
https://github.com/zephyrproject-rtos/zephyr/pull/41923/commits/14279d10e89abad9574929a0b710c3ab0ed90dff

File: nrf_security/Kconfig.tls [hunk 7]
```diff
@@ -0,0 +1,283 @@
...
+	range 0 16384
+	default 900 if OPENTHREAD_NRF_SECURITY
+	default 16380
+
+config MBEDTLS_SSL_IN_CONTENT_LEN
+	prompt "Max length for TLS outgoing fragments"
+	int
+	range 0 16384
+	default 900 if OPENTHREAD_NRF_SECURITY
+	default 16380
```
Comment (AndreyDodonov-EH) [inline nrf_security/Kconfig.tls:281]: This constant (16380) leads to mbedtls not working after this fix https://github.com/nrfconnect/sdk-mbedtls/commit/668b31f210356ef26b1ca26bced775b3d36f63f9

(unless of course server sends less)

Glad this config and wrong constant is removed altogether now.

But with SDK-NRF 2.4.2 it's not working...

**Pull request merged**